src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,342 @@
+    {
+      MDV_observed.push_back(it->getMetaValue("peak_apex_int"));
+    }
+    
+    double corrected_value;
+    corrected_feature = normalized_feature;
+    for (size_t i = 0; i < correction_matrix_inversed.size(); ++i) {
+      corrected_value = 0.0;
+      for (size_t j = 0; j < correction_matrix_inversed[0].size(); ++j) {
+        corrected_value += correction_matrix_inversed[i][j] * MDV_observed[j];
+      }
+      corrected_feature.getSubordinates().at(i).setIntensity(corrected_value);
+    }
+  }
+
+  void IsotopeLabelingMDVs::isotopicCorrections(FeatureMap& normalized_featureMap, FeatureMap& corrected_featureMap, std::vector<std::vector<double>> correction_matrix)
+  {
+    for (FeatureMap::Iterator feature_it = normalized_featureMap.begin(); feature_it != normalized_featureMap.end(); feature_it++){
+      Feature corrected_feature;
+      isotopicCorrection( *feature_it, corrected_feature, correction_matrix);
+      corrected_featureMap.push_back(corrected_feature);
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateIsotopicPurity(
+    Feature& normalized_featuremap,
+    Feature& featuremap_with_isotopic_purity,
+    std::vector<double>& experiment_data,
+    std::string& isotopic_purity_name)
+  {
+    featuremap_with_isotopic_purity = normalized_featuremap;
+    
+    double experiment_data_peak = 0.0;
+    if ( !experiment_data.empty() )
+    {
+      std::vector<double>::iterator max_it  = std::max_element(experiment_data.begin(), experiment_data.end());
+      uint64_t experiment_data_peak_idx     = std::distance(experiment_data.begin(), max_it);
+      experiment_data_peak                  = experiment_data[experiment_data_peak_idx];
+      
+      if ( experiment_data_peak_idx >= 1 && experiment_data_peak != 0.0)
+      {
+        double previous_experiment_data_peak  = experiment_data[experiment_data_peak_idx - 1];
+        double isotopic_purity                = experiment_data_peak_idx / (experiment_data_peak_idx + ( previous_experiment_data_peak / experiment_data_peak));
+        featuremap_with_isotopic_purity.setMetaValue(isotopic_purity_name, isotopic_purity);
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateIsotopicPurities(
+    FeatureMap& normalized_featureMap,
+    FeatureMap& featureMap_with_isotopic_purity,
+    std::vector<double>& experiment_data,
+    std::string& isotopic_purity_name)
+  { 
+    for (FeatureMap::Iterator feature_it = normalized_featureMap.begin(); feature_it != normalized_featureMap.end(); feature_it++){
+      Feature feature_with_isotopic_purity;
+      calculateIsotopicPurity( *feature_it, feature_with_isotopic_purity, experiment_data, isotopic_purity_name);
+      featureMap_with_isotopic_purity.push_back(feature_with_isotopic_purity);
+    }
+  }
+
+  
+  void IsotopeLabelingMDVs::calculateMDVAccuracy(
+    Feature& normalized_feature,
+    Feature& feature_with_accuracy_info,
+    std::vector<double>& fragment_isotopomer_measured,
```

Comment:
please check all methods if parameters can be passed as const references otherwise a user might expect that this is an output parameter

Reply:
Thanks for your feedback, I have updated all parameter to be passed as const reference wherever necessary